File: code/demonstration/receiver/AndroidIteration3/app/src/main/java/de/fzi/sensidl/receiver/android/iteration3/sensor/Sensors.java
package de.fzi.sensidl.receiver.android.iteration3.sensor;

import java.util.ArrayList;
import java.util.List;

/**
 * An object that holds all sensors from the ti sensor tag
 */
public class Sensors {

	private List<Sensor> sensorList;

	public Sensors() {
		sensorList = new ArrayList<>();
		sensorList.add(new Sensor(SensorUUID.IR_TEMPERATURE_SERVICE, SensorUUID.IR_TEMPERATURE_DATA, SensorUUID.IR_TEMPERATURE_CONFIG));
		sensorList.add(new Sensor(SensorUUID.ACCELEROMETER_SERVICE, SensorUUID.ACCELEROMETER_DATA, SensorUUID.ACCELEROMETER_CONFIG));
		sensorList.add(new Sensor(SensorUUID.HUMIDITY_SERVICE, SensorUUID.HUMIDITY_DATA, SensorUUID.HUMIDITY_CONFIG));
		sensorList.add(new Sensor(SensorUUID.MAGNETOMETER_SERVICE, SensorUUID.MAGNETOMETER_DATA, SensorUUID.MAGNETOMETER_CONFIG));
	}

	public List<Sensor> getSensorList() {
		return sensorList;
	}
}

File: code/demonstration/receiver/XamarinIteration3/XamarinStufe3/XamarinStufe3/Sensors.cs
using System.Collections.Generic;

namespace XamarinIteration3
{
	// <summary>
	//  An object that holds all sensors from the ti sensor tag
	// </summary>
	public class Sensors
	{
		public List<Sensor> sensorList { get; }

		public Sensors ()
		{
			sensorList = new List<Sensor> ();
			sensorList.Add (new Sensor (SensorGuid.IR_TEMPERATURE_SERVICE, SensorGuid.IR_TEMPERATURE_DATA, SensorGuid.IR_TEMPERATURE_CONFIG));
			sensorList.Add (new Sensor (SensorGuid.ACCELEROMETER_SERVICE, SensorGuid.ACCELEROMETER_DATA, SensorGuid.ACCELEROMETER_CONFIG));
			sensorList.Add (new Sensor (SensorGuid.HUMIDITY_SERVICE, SensorGuid.HUMIDITY_DATA, SensorGuid.HUMIDITY_CONFIG));
			sensorList.Add (new Sensor (SensorGuid.MAGNETOMETER_SERVICE, SensorGuid.MAGNETOMETER_DATA, SensorGuid.MAGNETOMETER_CONFIG));
		}
	}
}
